src/utils/thread_helper.cpp

```diff
@@ -1328,6 +1346,10 @@ otError ThreadHelper::RetrieveTelemetryData(Mdns::Publisher *aPublisher, threadn
             srpServerResponseCounters->set_name_exists_count(responseCounters->mNameExists);
             srpServerResponseCounters->set_refused_count(responseCounters->mRefused);
             srpServerResponseCounters->set_other_count(responseCounters->mOther);
+#if OT_DNS_UPSTREAM_QUERY
+            srpServer->set_upstream_dns_query_state(
+                UpstreamDnsQueryStateFromOtDnssdUpstreamQuery(otDnssdUpstreamQueryIsEnabled(mInstance)));
```

Comment:
Since the logic is very simple, how about

```suggestion
            srpServer->set_upstream_dns_query_state(otDnssdUpstreamQueryIsEnabled(mInstance) ? threadnetwork::TelemetryData::UPSTREAMDNS_QUERY_STATE_ENABLED : threadnetwork::TelemetryData::UPSTREAMDNS_QUERY_STATE_DISABLED);
```

Reply:
Thanks!